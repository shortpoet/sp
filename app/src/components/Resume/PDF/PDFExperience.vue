<template>
  <section :class="sectionClass" id="experience">
    <div class="my-auto">
      <h4 class="heading heading-4" style="font-family: 'Saira Extra Condensed';">Experience</h4>
      <div class="d-flex flex-column justify-content-between" v-for="type in experiences" :key="type.type">
        <!-- <h4 class="pdf-subheading mt-2 mb-0">{{ type.type }} Experience</h4> -->
        <div v-for="(job, ji) in type.jobs" :key="ji">
          <div class="d-flex flex-column justify-content-between">
            <div class="d-flex flex-row">
              <div class="mr-auto">
                <h5 id="job-position" style="font-family: 'Saira Extra Condensed';" class="heading heading-5 mb-0">{{ job.position }}</h5>
              </div>
              <div :class="textRightClass">
                <h6 style="font-family: 'Saira Extra Condensed';" class="heading heading-6 pdf-experience-dates text-primary">{{ job.startDate }} - {{ job.endDate }}</h6>
              </div>
            </div>
            <div class="d-flex flex-column">
              <h6 style="font-family: 'Saira Extra Condensed';" class="heading heading-6 pdf-company">{{ job.company }}</h6>
              <div
                v-for="(text, di) in job.description.split('\\n\\r')"
                :key="di"
                class="mb-2"
              >
                <p style="font-family: 'Open Sans';" v-html="text" :class="'job-' + (ji + 1) + '-description-' + (di + 1)"></p>
              </div>

            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</template>
<script>
export default {
  name: 'PDFExperience',
  props: {
    experiences: {
      type: Array
    },
    renderPDF: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    sectionClass () {
      return this.renderPDF ?
        'resume-section-render ml-3 mt-2 mb-3'
        :
        'resume-section ml-3 mt-2 mb-3'
    },
    textRightClass () {
      return this.renderPDF ?
        'resume-date text-md-right mr-1'
        :
        'resume-date text-md-right mr-4'
    }
  }
}
</script>
