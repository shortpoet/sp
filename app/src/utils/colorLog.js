const chalk = require('chalk')
import {inspect} from 'util'

// const _log  = () => Function.prototype.bind.call(console.log, console, 'LOG_ALIAS')

export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
