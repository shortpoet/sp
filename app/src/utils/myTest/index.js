import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
import propFinder from './test.utils.propFinder'
import propsFinder from './test.utils.propsFinder'
import propMocker from './test.utils.propMocker'
import propsMocker from './test.utils.propsMocker'
import textMatcher from './test.utils.textMatcher'
import textMatcherFactory from './test.utils.textMatcherFactory'

export default {
  createWrapper,
  factory,
  factoryShallow,
  propFinder,
  propsFinder,
  propMocker,
  propsMocker,
  textMatcher,
  textMatcherFactory
}
