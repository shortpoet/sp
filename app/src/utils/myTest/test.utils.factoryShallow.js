import { shallowMount } from '@vue/test-utils'

export const factoryShallow = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    return shallowMount(component, options)
  }

module.exports = factoryShallow