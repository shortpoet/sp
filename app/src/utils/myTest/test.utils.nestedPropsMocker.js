

// // this is incomplete
// // maybe try and develop using TDD  
  
// export const nestedPropsMocker = (props) => {

//   console.log(props)

//   let propsData = {}


//   const handleArray = (props) => {
//     propsData = {}
//     props.map(prop => {
//       propsData[`${prop}`] = `sample ${prop}`
//     })
//     return {
//       propsData
//     }
//   }


//   const handleNested = (props, propsData, previousKey, nextKey) => {
//     console.log(props, propsData)

//     let firstLevel
//     if (previousKey === null && nextKey === null) {
//       firstLevel = true
//     }

//     // spread to clone so the splice works on subsequent tests
//     let values = [...props]

//     const length = parseInt(values.splice(-1, 1)[0])
//     console.log(length)
//     const iterator = [...Array(length).keys()]

//     if (firstLevel === true) {
//       previousKey = values.splice(0, 1)[0]
//       propsData[`${previousKey}`] = []
//     } else {
//       console.log('#### not first run ####')
//       console.log(propsData)
//       console.log(previousKey)
//       nextKey = values.splice(0, 1)[0]
//       console.log(nextKey)
//       propsData[`${previousKey}`][`${nextKey}`] = []
//     }


//     let keyDict

//     for (let i of iterator) {
//       keyDict = {}
//       values.map(prop => {
//         if (Array.isArray(prop)) {
//           console.log(prop)
//           console.log('#### calling handleNested from nested ####')
//           // this would only work for one level nested
//           // TODO maybe refactor for infinite recursion
//           console.log(previousKey)
//           console.log(nextKey)
//           console.log(propsData)
//           handleNested(prop, propsData, previousKey, nextKey)
//         } else {
//           console.log(prop)
//           keyDict[`${prop}`] = `sample ${prop}`
//         }
//       })
//       if (firstLevel === true) {
//         propsData[`${previousKey}`].push(keyDict)
//         firstRun = false
//       } else {
//         console.log(propsData)
//         console.log(previousKey)
//         console.log(nextKey)
//         console.log(propsData.experiences)
//         console.log(propsData.experiences.jobs)
//         propsData[`${previousKey}`][`${nextKey}`].push(keyDict)
//       }
//     }
//     return {
//       propsData
//     }
//   }

//   const isNested = props.some(p => Array.isArray(p))
//   const previousKey = null
//   const nextKey = null

//   if (isNested) {
//     console.log('#### is nested ####')
//     handleNested(props, propsData, previousKey, nextKey)
//   } else  {
//     handleArray(props)
//   }
// }
