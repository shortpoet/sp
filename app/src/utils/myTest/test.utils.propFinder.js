import {hardResume} from '@/assets/resume.js'

export const propFinder = (prop) => {
  return {
    propsData: {
      [`${prop}`]: hardResume[`${prop}`]
    }
  }
}

module.exports = propFinder