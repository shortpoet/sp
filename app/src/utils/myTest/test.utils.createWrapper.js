import { createLocalVue } from '@vue/test-utils'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'

// import VueRouter from 'vue-router'
import Vuex from 'vuex'
import PortalVue from 'portal-vue'

import createStore from '@/store/createStore'
// import createMockRouterConfig from '@/router/createMockRouterConfig'
// import paths from '@/router/paths.js'

import { log } from '@/utils/colorLog'
export const createWrapper  = (
  component,
  /* istanbul ignore next */
  options = {},
  // but why don't i have to ignore the empty  or even unprovided resumeStoreOtions
  // might need to figure this out but setting params to ignore for now
  resumeStoreOptions = {},
  // routerOptions = {},
  isShallow = false,
  ) => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(PortalVue)
    const mockStoreResume = createStore.createMocks().createStoreResumeMocks(resumeStoreOptions)
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})

    // Using router is untested makes tests fail
    // and just doesn't work for now
    // localVue.use(VueRouter)
    // const routerConfig = createMockRouterConfig.createMockRouterConfig(routerOptions)
    // const router = new VueRouter(routerConfig)

    // log('green', mockStoreResume)
    // log('yellow', storeConfig.modules.resume)
    const store = new Vuex.Store(storeConfig)
    // log('red', store)
    // log('blue', store._mutations['resume/SET_RESUME_RAW'][0]())
    // console.log(options)
    return isShallow ? 
    factoryShallow(component, {
      localVue,
      store,
      // router,
      ...options
    }) : 
    factory(component, {
      localVue,
      store,
      // router,
      ...options
    })
}

// export default createWrapper
// when this was set to export default a console log(createWrapper) that was in
// test.utils.textMatcher line 19 was being executed from every module that called createWrapper
module.exports = createWrapper