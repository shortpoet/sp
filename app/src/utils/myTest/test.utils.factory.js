import { mount } from '@vue/test-utils'
import { log } from '@/utils/colorLog'
export const factory = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    // this only breaks when testing
    // probably should comment out of real func (which this is (now) hehe)
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    // console.log(options)
    // console.log(options.mocks)
    // console.log(options.store)
    // console.log(options.localVue)
    // log('blue', options.stubs)
    // log('green', options.propsData)
    // console.log(mount)
    return mount(component, options)
  }

module.exports = factory

// export default factory