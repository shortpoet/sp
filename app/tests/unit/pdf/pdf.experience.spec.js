import * as myTest from '@/utils/myTest'
const { createWrapper, propsMocker } = myTest.default
import PDFExperience from '@/components/Resume/PDF/PDFExperience'
import {cloneDeep} from 'lodash'

describe('PDFExperience.vue', () => {

  const component = PDFExperience
  // const props = Object.keys(component.props)

  // this was for the _propsMocker attempt
  // const props = [
  //     'experiences',
  //     'type',
  //     ['jobs', 'position', 'company', 'description', '1'],
  //     '4'
  //   ]

  const experiences = ['software', 'language', 'sales', 'hospitality']

  const jobs =  [propsMocker(['position', 'company', 'description', 'startDate', 'endDate']).propsData]
  const propsObject = {
    propsData: {
      experiences: experiences.map(e => {
        return {
          type: e,
          jobs: jobs
        }
      })
    }
  }
  let wrapper = createWrapper(component, propsObject)

  let prop
  let selector
  let mockProp = true

  beforeEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('renders experience h4 header', () => {

    prop = 'Experience'
    selector = 'h4'
    expect(wrapper.find(`${selector}`).text()).toMatch(prop)

  })
  it('renders experience h5 that matches position prop', () => {

    prop = 'sample position'
    selector = 'h5'
    expect(wrapper.find(`${selector}`).text()).toMatch(prop)

  })
  it('renders experience .pdf-company that matches company prop', () => {

    prop = 'sample company'
    selector = '.pdf-company'
    expect(wrapper.find(`${selector}`).text()).toMatch(prop)

  })
  it('renders experience .pdf-experience-dates that matches start/end date prop', () => {

    prop = 'sample startDate - sample endDate'
    selector = '.pdf-experience-dates'
    expect(wrapper.find(`${selector}`).text()).toMatch(prop)

  })
  it('renders experience p that matches text prop accounting for splits', () => {

    let _propsObject = cloneDeep(propsObject)
    _propsObject.propsData.experiences.map(e => {
      e.jobs.map(j => {
        j.description = `description part 1\\n\\rdescription part 2\\n\\rdescription part 3`
      })
    })    
    
    selector = 'p'
    wrapper = createWrapper(component, _propsObject)

    const typeArray = wrapper.findAll(`${selector}`)

    let pIndex = 0
    _propsObject.propsData.experiences.forEach((e, ie) => {
      e.jobs.forEach((j, ij) => {
          for (let id in [...Array(3).keys()]) {
            id = parseInt(id)
            const hashedClass = `job-${ij + 1}-description-${id + 1}`
            expect(typeArray.at(pIndex).attributes().class).toMatch(hashedClass)
            pIndex++
          }
        })
    })

  })
  it('matches experience snapshot', () => {
    
    // same wrapper object from previous tests adds the split descriptions
    // bec a new wrapper was not instantiated

    expect(wrapper.html()).toMatchSnapshot()


  })
  it('matches renderPDF experience snapshot', () => {
    
    // new wrapper bec need to modify propsData to add renderPDF boolean
    
    let _propsObject = cloneDeep(propsObject)
    _propsObject.propsData.experiences.map(e => {
      e.jobs.map(j => {
        j.description = `description part 1\\n\\rdescription part 2\\n\\rdescription part 3`
      })
    })    
    _propsObject.propsData.renderPDF = true

    wrapper = createWrapper(component, _propsObject)

    expect(wrapper.html()).toMatchSnapshot()

  })
})
