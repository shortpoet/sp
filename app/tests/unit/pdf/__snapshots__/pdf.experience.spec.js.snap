// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFExperience.vue matches experience snapshot 1`] = `
<section id="experience" class="resume-section ml-3 mt-2 mb-3">
  <div class="my-auto">
    <h4 class="heading heading-4" style="font-family: 'Saira Extra Condensed';">Experience</h4>
    <div class="d-flex flex-column justify-content-between">
      <div>
        <div class="d-flex flex-column justify-content-between">
          <div class="d-flex flex-row">
            <div class="mr-auto">
              <h5 id="job-position" class="heading heading-5 mb-0" style="font-family: 'Saira Extra Condensed';">sample position</h5>
            </div>
            <div class="resume-date text-md-right mr-4">
              <h6 class="heading heading-6 pdf-experience-dates text-primary" style="font-family: 'Saira Extra Condensed';">sample startDate - sample endDate</h6>
            </div>
          </div>
          <div class="d-flex flex-column">
            <h6 class="heading heading-6 pdf-company" style="font-family: 'Saira Extra Condensed';">sample company</h6>
            <div class="mb-2">
              <p class="job-1-description-1" style="font-family: 'Open Sans';">description part 1</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-2" style="font-family: 'Open Sans';">description part 2</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-3" style="font-family: 'Open Sans';">description part 3</p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="d-flex flex-column justify-content-between">
      <div>
        <div class="d-flex flex-column justify-content-between">
          <div class="d-flex flex-row">
            <div class="mr-auto">
              <h5 id="job-position" class="heading heading-5 mb-0" style="font-family: 'Saira Extra Condensed';">sample position</h5>
            </div>
            <div class="resume-date text-md-right mr-4">
              <h6 class="heading heading-6 pdf-experience-dates text-primary" style="font-family: 'Saira Extra Condensed';">sample startDate - sample endDate</h6>
            </div>
          </div>
          <div class="d-flex flex-column">
            <h6 class="heading heading-6 pdf-company" style="font-family: 'Saira Extra Condensed';">sample company</h6>
            <div class="mb-2">
              <p class="job-1-description-1" style="font-family: 'Open Sans';">description part 1</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-2" style="font-family: 'Open Sans';">description part 2</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-3" style="font-family: 'Open Sans';">description part 3</p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="d-flex flex-column justify-content-between">
      <div>
        <div class="d-flex flex-column justify-content-between">
          <div class="d-flex flex-row">
            <div class="mr-auto">
              <h5 id="job-position" class="heading heading-5 mb-0" style="font-family: 'Saira Extra Condensed';">sample position</h5>
            </div>
            <div class="resume-date text-md-right mr-4">
              <h6 class="heading heading-6 pdf-experience-dates text-primary" style="font-family: 'Saira Extra Condensed';">sample startDate - sample endDate</h6>
            </div>
          </div>
          <div class="d-flex flex-column">
            <h6 class="heading heading-6 pdf-company" style="font-family: 'Saira Extra Condensed';">sample company</h6>
            <div class="mb-2">
              <p class="job-1-description-1" style="font-family: 'Open Sans';">description part 1</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-2" style="font-family: 'Open Sans';">description part 2</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-3" style="font-family: 'Open Sans';">description part 3</p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="d-flex flex-column justify-content-between">
      <div>
        <div class="d-flex flex-column justify-content-between">
          <div class="d-flex flex-row">
            <div class="mr-auto">
              <h5 id="job-position" class="heading heading-5 mb-0" style="font-family: 'Saira Extra Condensed';">sample position</h5>
            </div>
            <div class="resume-date text-md-right mr-4">
              <h6 class="heading heading-6 pdf-experience-dates text-primary" style="font-family: 'Saira Extra Condensed';">sample startDate - sample endDate</h6>
            </div>
          </div>
          <div class="d-flex flex-column">
            <h6 class="heading heading-6 pdf-company" style="font-family: 'Saira Extra Condensed';">sample company</h6>
            <div class="mb-2">
              <p class="job-1-description-1" style="font-family: 'Open Sans';">description part 1</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-2" style="font-family: 'Open Sans';">description part 2</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-3" style="font-family: 'Open Sans';">description part 3</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`PDFExperience.vue matches renderPDF experience snapshot 1`] = `
<section id="experience" class="resume-section-render ml-3 mt-2 mb-3">
  <div class="my-auto">
    <h4 class="heading heading-4" style="font-family: 'Saira Extra Condensed';">Experience</h4>
    <div class="d-flex flex-column justify-content-between">
      <div>
        <div class="d-flex flex-column justify-content-between">
          <div class="d-flex flex-row">
            <div class="mr-auto">
              <h5 id="job-position" class="heading heading-5 mb-0" style="font-family: 'Saira Extra Condensed';">sample position</h5>
            </div>
            <div class="resume-date text-md-right mr-1">
              <h6 class="heading heading-6 pdf-experience-dates text-primary" style="font-family: 'Saira Extra Condensed';">sample startDate - sample endDate</h6>
            </div>
          </div>
          <div class="d-flex flex-column">
            <h6 class="heading heading-6 pdf-company" style="font-family: 'Saira Extra Condensed';">sample company</h6>
            <div class="mb-2">
              <p class="job-1-description-1" style="font-family: 'Open Sans';">description part 1</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-2" style="font-family: 'Open Sans';">description part 2</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-3" style="font-family: 'Open Sans';">description part 3</p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="d-flex flex-column justify-content-between">
      <div>
        <div class="d-flex flex-column justify-content-between">
          <div class="d-flex flex-row">
            <div class="mr-auto">
              <h5 id="job-position" class="heading heading-5 mb-0" style="font-family: 'Saira Extra Condensed';">sample position</h5>
            </div>
            <div class="resume-date text-md-right mr-1">
              <h6 class="heading heading-6 pdf-experience-dates text-primary" style="font-family: 'Saira Extra Condensed';">sample startDate - sample endDate</h6>
            </div>
          </div>
          <div class="d-flex flex-column">
            <h6 class="heading heading-6 pdf-company" style="font-family: 'Saira Extra Condensed';">sample company</h6>
            <div class="mb-2">
              <p class="job-1-description-1" style="font-family: 'Open Sans';">description part 1</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-2" style="font-family: 'Open Sans';">description part 2</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-3" style="font-family: 'Open Sans';">description part 3</p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="d-flex flex-column justify-content-between">
      <div>
        <div class="d-flex flex-column justify-content-between">
          <div class="d-flex flex-row">
            <div class="mr-auto">
              <h5 id="job-position" class="heading heading-5 mb-0" style="font-family: 'Saira Extra Condensed';">sample position</h5>
            </div>
            <div class="resume-date text-md-right mr-1">
              <h6 class="heading heading-6 pdf-experience-dates text-primary" style="font-family: 'Saira Extra Condensed';">sample startDate - sample endDate</h6>
            </div>
          </div>
          <div class="d-flex flex-column">
            <h6 class="heading heading-6 pdf-company" style="font-family: 'Saira Extra Condensed';">sample company</h6>
            <div class="mb-2">
              <p class="job-1-description-1" style="font-family: 'Open Sans';">description part 1</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-2" style="font-family: 'Open Sans';">description part 2</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-3" style="font-family: 'Open Sans';">description part 3</p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="d-flex flex-column justify-content-between">
      <div>
        <div class="d-flex flex-column justify-content-between">
          <div class="d-flex flex-row">
            <div class="mr-auto">
              <h5 id="job-position" class="heading heading-5 mb-0" style="font-family: 'Saira Extra Condensed';">sample position</h5>
            </div>
            <div class="resume-date text-md-right mr-1">
              <h6 class="heading heading-6 pdf-experience-dates text-primary" style="font-family: 'Saira Extra Condensed';">sample startDate - sample endDate</h6>
            </div>
          </div>
          <div class="d-flex flex-column">
            <h6 class="heading heading-6 pdf-company" style="font-family: 'Saira Extra Condensed';">sample company</h6>
            <div class="mb-2">
              <p class="job-1-description-1" style="font-family: 'Open Sans';">description part 1</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-2" style="font-family: 'Open Sans';">description part 2</p>
            </div>
            <div class="mb-2">
              <p class="job-1-description-3" style="font-family: 'Open Sans';">description part 3</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
